src/plugins/visualizations/public/vis_types/base_vis_type.ts

```diff
@@ -17,118 +17,113 @@
-  name: string;
-  title: string;
-  description: string;
-  getSupportedTriggers?: () => Array<keyof TriggerContextMapping>;
-  icon?: string;
-  image?: string;
-  stage: 'experimental' | 'beta' | 'production';
-  isExperimental: boolean;
-  options: Record<string, any>;
-  visualization: VisualizationControllerConstructor | undefined;
-  visConfig: Record<string, any>;
-  editor: any;
-  editorConfig: Record<string, any>;
-  hidden: boolean;
-  requiresSearch: boolean;
-  requestHandler: string | unknown;
-  responseHandler: string | unknown;
-  hierarchicalData: boolean | unknown;
-  setup?: unknown;
-  useCustomNoDataScreen: boolean;
-  inspectorAdapters?: Adapters | (() => Adapters);
-  toExpressionAst?: VisToExpressionAst<TVisParams>;
-  getDeprecationMessage?: (vis: Vis) => ReactElement<{}>;
+const defaultOptions: VisTypeOptions = {
+  showTimePicker: true,
+  showQueryBar: true,
+  showFilterBar: true,
+  showIndexSelection: true,
+  hierarchicalData: false, // we should get rid of this i guess ?
+};
+
+export class BaseVisType<TVisParams = VisParams> implements VisType<TVisParams> {
+  public readonly name;
+  public readonly title;
+  public readonly description;
+  public readonly getSupportedTriggers;
+  public readonly icon;
+  public readonly image;
+  public readonly stage;
+  public readonly options;
+  public readonly visualization;
+  public readonly visConfig;
+  public readonly editor;
+  public readonly editorConfig;
+  public hidden;
+  public readonly requestHandler;
+  public readonly responseHandler;
+  public readonly hierarchicalData;
+  public readonly setup;
+  public readonly useCustomNoDataScreen;
+  public readonly inspectorAdapters;
+  public readonly toExpressionAst;
+  public readonly getDeprecationMessage;
 
   constructor(opts: BaseVisTypeOptions<TVisParams>) {
     if (!opts.icon && !opts.image) {
       throw new Error('vis_type must define its icon or image');
     }
 
-    const defaultOptions = {
-      // controls the visualize editor
-      showTimePicker: true,
-      showQueryBar: true,
-      showFilterBar: true,
-      showIndexSelection: true,
-      hierarchicalData: false, // we should get rid of this i guess ?
-    };
-
     this.name = opts.name;
-    this.description = opts.description || '';
+    this.description = opts.description ?? '';
     this.getSupportedTriggers = opts.getSupportedTriggers;
     this.title = opts.title;
     this.icon = opts.icon;
     this.image = opts.image;
     this.visualization = opts.visualization;
-    this.visConfig = _.defaultsDeep({}, opts.visConfig, { defaults: {} });
+    this.visConfig = defaultsDeep({}, opts.visConfig, { defaults: {} });
     this.editor = opts.editor;
-    this.editorConfig = _.defaultsDeep({}, opts.editorConfig, { collections: {} });
-    this.options = _.defaultsDeep({}, opts.options, defaultOptions);
-    this.stage = opts.stage || 'production';
-    this.isExperimental = opts.stage === 'experimental';
```

Comment:
:information_source: The `isExperimental` flag was only used in one place. I just inlined the check against `stage` there, so keep the `VisType` interface a bit smaller.